### ISSUE TITLE
Bug: minifier doesn't work without assets/vendor

### ISSUE THREAD

**Issue by snayam** [sensiolabs/minify-bundle#23]

**Comment by snayam** [comment]
I use asset-mapper but without third-party packages, so i removed the assets/vendor folder. In this case, files are not minified after asset-map:compile.

To work, i have to 2 solutions :

1. Create empty asset/vendor folder, or 
2. Set ignore_vendor to false in sensiolabs_minify config

**Comment by smnandre** [comment]
hi @snayam !
I just pushed a fix and deployed a new version, could you check and tell me if it solves your problem?

**Comment by snayam** [comment]
Hi @smnandre 

I've just tried but same problem :-(

assets/vendor folder present => asset-map:compile -> minifier works fine !
assets/vendor folder not present  => asset-map:compile -> minifier doesn't work

In logs, i have the following php debug message, but display in the 2 cases so i don't think it's related : 

`php.DEBUG: Warning: mkdir(): File exists {"exception":{"Symfony\\Component\\ErrorHandler\\Exception\\SilencedErrorContext":{"severity":2,"file":"/vendor/symfony/asset-mapper/CompiledAssetMapperConfigReader.php","line":43,"trace":[{"file":"/vendor/symfony/asset-mapper/Command/AssetMapperCompileCommand.php","line":78,"function":"saveConfig","class":"Symfony\\Component\\AssetMapper\\CompiledAssetMapperConfigReader","type":"->"}],"count":1}}} []`

Symfony 7.1.8 / PHP 8.2.25

**Comment by smnandre** [comment]
Could you, just to be sure, run this command ? 

```
php bin/console c:c && php bin/console debug:container --tag asset_mapper
```

> assets/vendor folder present => asset-map:compile -> minifier works fine !
assets/vendor folder not present => asset-map:compile -> minifier doesn't work

Same thing, just to be sure, you do clear (`bin/console c:p:c --all`) and reset assets (`rm -rf public/assets`) between these tests ?

**Comment by snayam** [comment]
> Could you, just to be sure, run this command ?
> 
> ```
> php bin/console c:c && php bin/console debug:container --tag asset_mapper
> ```

<img width="1296" alt="Capture d’écran 2024-11-25 à 09 22 12" src="https://github.com/user-attachments/assets/0cb517cd-fda9-438a-b33d-ce78d2be815a">

- 

> Same thing, just to be sure, you do clear (`bin/console c:p:c --all`) and reset assets (`rm -rf public/assets`) between these tests ?

Done but same problem.

I think the issue comes from supports function (MinifierCompiler.php). 
Here, $asset->isVendor return **true** when assets/vendor is present ; **false** when is not present.

```
public function supports(MappedAsset $asset): bool
    {
        if (!in_array($asset->publicExtension, $this->extensions, true)) {
            return false;
        }

        if ($this->ignoreVendor && $asset->isVendor) {
            return false;
        }

        foreach ($this->ignorePaths as $ignorePath) {
            if (fnmatch($ignorePath, $asset->sourcePath)) {
                return false;
            }
        }

        return true;
    }
```

**Comment by smnandre** [comment]
>   Here, $asset->isVendor return true when assets/vendor is present ; false when is not present.

For wich asset ? One in particular or all of them ?

**Comment by smnandre** [comment]
I think this is due to this bug (fixed and released in the next version) ..  not something related to this bundle :/ 

https://github.com/symfony/symfony/commit/9e3984ff93fdb1883bbb289f40da2f503439b023

**Comment by smnandre** [comment]
Did you check if Symfony 7.2 solves your issue @snayam ?

**Comment by smnandre** [comment]
Closing here as it seems fixed in Symfony, just need to wait. Please reopen if you need :)

**Comment by snayam** [comment]
Hi @smnandre,
Sorry for my late reply. I confirm that the issue is solved with symfony 7.2.
Thanks ;-)
